src/Npgsql/NpgsqlConnection.cs

```diff
@@ -717,10 +729,11 @@ internal Task Close(bool async, CancellationToken cancellationToken = default)
                 // TODO: Consider falling through to the regular reset logic. This adds some unneeded conditions
                 // and assignment but actual perf impact should be negligible (measure).
                 Debug.Assert(Connector == null);
+                Volatile.Write(ref _closing, 0);
```

Comment:
To not end with a broken state due to some failure in user code. This is how we do it in other places, buy the way.

Reply:
Do you mean because of an exception raised from the event handler? If that's the goal, then I think it's better to catch and swallow that exception instead. It seems safer to defer resetting _closing to 0 until everything is done (I'm thinking specifically of a scenario where the user's event callback somehow closes again).

> This is how we do it in other places, buy the way.

Can you point to an example?

Reply:
For example, logging and changing the state happens after internal parts [here](https://github.com/npgsql/npgsql/blob/98d53eadb9ba3ad8e0e5cdffb514a1913a5718d5/src/Npgsql/NpgsqlConnection.cs#L801).

Reply:
Logging is quite different, _closing is a safety flag to make sure we don't get into a bad state.

Swallowing any exceptions coming out of the user code needs to be done in any case, otherwise they may screw up Npgsql's internal flow when the event is triggered from Break.